src/serializer/functions.js

```diff
@@ -255,29 +255,39 @@ export class Functions {
     return noOpFunc;
   }
 
+  _forEachBindingOfEffects(effects: Effects, func: (binding: Binding) => void): void {
+    let [result, , nestedBindingsToIgnore] = effects.data;
+    for (let [binding] of nestedBindingsToIgnore) {
+      func(binding);
+    }
+    if (result instanceof PossiblyNormalCompletion) {
+      this._forEachBindingOfEffects(result.alternateEffects, func);
+      this._forEachBindingOfEffects(result.consequentEffects, func);
+    }
```

Comment:
What about other completion types like `JoinedAbruptCompletions` (which also have `consequentEffects` and `alternateEffects`? Why or why not do we need to handle them here?